packages/serverpod_serialization/test/websocket_messages_test.dart

```diff
@@ -1,29 +1,106 @@
-import 'package:serverpod_serialization/src/websocket_messages.dart';
+import 'package:serverpod_serialization/serverpod_serialization.dart';
 import 'package:test/test.dart';
-import 'package:uuid/uuid.dart';
+
+class _TestSerializationManager extends SerializationManager {
+  _TestSerializationManager();
+  @override
+  dynamic deserializeByClassName(Map<String, dynamic> data) {
+    if (data['className'] == '_SimpleData') {
+      return deserialize<_SimpleData>(data['data']);
+    }
+    if (data['className'] == '_TestSerializableException') {
+      return deserialize<_TestSerializableException>(data['data']);
+    }
+
+    return super.deserializeByClassName(data);
+  }
+
+  @override
+  String? getClassNameForObject(Object? data) {
+    String? className = super.getClassNameForObject(data);
+    if (className != null) return className;
+    if (data is _SimpleData) {
+      return '_SimpleData';
+    }
+    if (data is _TestSerializableException) {
+      return '_TestSerializableException';
+    }
+    return null;
+  }
+
+  @override
+  T deserialize<T>(
+    dynamic data, [
+    Type? t,
+  ]) {
+    t ??= T;
+    if (t == _SimpleData) {
+      return _SimpleData.fromJson(data) as T;
+    }
+    if (t == _TestSerializableException) {
+      return _TestSerializableException() as T;
+    }
+
+    return super.deserialize<T>(data, t);
+  }
+}
+
+class _SimpleData implements SerializableModel {
+  final String data;
+
+  _SimpleData(this.data);
+
+  @override
+  Map<String, dynamic> toJson() {
+    return {
+      'data': data,
+    };
+  }
+
+  @override
+  factory _SimpleData.fromJson(Map<String, dynamic> json) {
+    return _SimpleData(json['data']);
+  }
+}
```

Comment:
Should add a server only field and validate that it is not included in the serialization.